### PR TITLE
#599 review home

### PR DESCRIPTION
**Pull request by nmadruga**
Still working on this...

There were more than I expected to be worked with, Row for assignment, Consolidation, reviews. 
I have looked into most of it I believe, that's why making this Draft PR, should be continuing to work a little more today to get all actions styled (and fixing a few issues with spacing of elements inside each row).

I think it might be good to merge as it is, and I would continue in another PR

### CODE REVIEW

File: semantic/src/site/collections/grid.overrides
Comment (CarlosNZ) [inline semantic/src/site/collections/grid.overrides:27]: Hey, I don't think it's a good idea to go over-riding the default styles for all Grids, since Grids are already in use in other places. Primarily, the main application form page from my last PR (#647) now has weirdly wide margins due to these changes, which makes the Progress bar look manky.

How it should be:

Screen Shot 2021-05-02 at 11 23 10 PM

How it is now:
Screen Shot 2021-05-02 at 11 24 13 PM


Probably the best thing to do is to use `#id`s on the specific elements you want to over-ride. Since grids are very much a "custom" thing that changes on a case-by-case basis, we should probably leave the defaults as is (unless you think there is a strong reason not to?)

File: src/components/Review/ReviewSectionRowAssigned.tsx
Comment (CarlosNZ) [inline src/components/Review/ReviewSectionRowAssigned.tsx:18]: Just wondering why this needs to be a "Label" component as opposed to just straight text? Maybe the formatting (i.e. vertical alignment) would work better for straight text. Doesn't seem like anything is really being "labelled".

File: semantic/src/site/elements/label.overrides
Comment (CarlosNZ) [inline semantic/src/site/elements/label.overrides:52]: You should be able to achieve this by just over-riding the variables, I don't think you need to override the class definitions. (since it's really just colours your re-mapping, right?)

Comment (CarlosNZ) [reply semantic/src/site/elements/label.overrides:52]: For example, see "Form.variables" -- there's a bunch of variables where I've just over-ridden the colour definitons to point to our colour palette.

File: semantic/src/site/elements/icon.overrides
Comment (CarlosNZ) [inline semantic/src/site/elements/icon.overrides:12]: (Similar to my comment below): You should be able to just over-ride variables to achieve this, no need to re-write class definitions.

File: semantic/src/site/globals/site.overrides
Comment (CarlosNZ) [inline semantic/src/site/globals/site.overrides:397]: Great, thanks for deleting this. I've been wanting to but wasn't sure it wasn't needed.

File: src/components/Application/ApplicationContainer.tsx
Comment (CarlosNZ) [inline src/components/Application/ApplicationContainer.tsx:19]: Why remove the id? I know there was no longer any CSS referring to it, but it's still helpful to have ids for thing, especially major structural components like this. (Makes it a lot easier seeing what's what in the Inspector and relating it back to the code).

File: src/components/Review/ReviewContainer.tsx
Comment (CarlosNZ) [inline src/components/Review/ReviewContainer.tsx:15]: Probably want to chuck in an id for this, since it's a fairly major element.

File: semantic/src/site/elements/label.overrides
Comment (CarlosNZ) [inline semantic/src/site/elements/label.overrides:34]: I don't think this needs its own class. To put a label in italics, just wrap it the text in `<em></em>` tags.

File: semantic/src/site/elements/label.overrides
Comment (CarlosNZ) [inline semantic/src/site/elements/label.overrides:38]: Wouldn't we just make a generic "info-text" class that can be used for ALL text elements that need to be coloured `@information` colour? No need to restrict it to one particular label type.

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
I'll have a closer look at this in the morn, I just highlighted a few things that I noticed from a quick look through.

Most important is the Grid over-rides. (see comments)

File: semantic/src/site/collections/grid.overrides
Comment (nmadruga) [reply semantic/src/site/collections/grid.overrides:27]: Sorry, I didn't realise it had changed the layout of the In progress page : (
I thought that the `border-radius: 6px` would be something global to all grids, but didn't realise the margin was going to be a bad decision. I'll add that to a specific `id`. All the other styles are under a **className** `.stripes`, so should be ok, right?

File: src/components/Review/ReviewSectionRowAssigned.tsx
Comment (nmadruga) [reply src/components/Review/ReviewSectionRowAssigned.tsx:18]: Can you give me an example please? I was using like that to keep all the Global layout we have set for Semantic-ui components, and using text, not wrapped with something seemed difficult to set styles, since there is no tags related to it.

File: semantic/src/site/collections/grid.overrides
Comment (CarlosNZ) [reply semantic/src/site/collections/grid.overrides:27]: > All the other styles are under a **className** `.stripes`, so should be ok, right?

Yeah, looks like it.

File: semantic/src/site/elements/label.overrides
Comment (nmadruga) [reply semantic/src/site/elements/label.overrides:52]: Sorry, end up not using this, using the `<Icon link ...>` styled seemed as a better approach. Will remove this unused definitions now.

File: src/components/Review/ReviewSectionRowAssigned.tsx
Comment (CarlosNZ) [reply src/components/Review/ReviewSectionRowAssigned.tsx:18]: Yeah okay, I see what you're saying.  "simple-label" works as an extension of "Label". Cool.

Just wondering what's causing the vertical alignment problem though. Looking at the `.simple-label` definition, I think the problem is the "height" specification. It's artificially forcing a height onto something whose height really should be determined by the text itself. If I remove `height: 20px;`, it correctly centers vertically. There's still an issue with the rows being taller than what you want (I think), but it looks better and it's more of an issue of wrangling margins and padding to make it narrower, I'd say.

File: semantic/src/site/collections/grid.overrides
Comment (nmadruga) [reply semantic/src/site/collections/grid.overrides]: I tried something else, have changed the margin inside each row, and started using the Grid.Row with this purpose on `ReviewSectionRow` and `AssignmentSectionRow`, both are contained in a semantic component (Segment) that is using the `.stripes` class. Hope that is a better solution to this: 32c935a

File: src/components/Review/ReviewSectionRowAssigned.tsx
Comment (nmadruga) [reply src/components/Review/ReviewSectionRowAssigned.tsx:18]: > Looking at the .simple-label definition, I think the problem is the "height" specification

Yes, I noticed that too, and have removed on commit 32c935a

File: src/components/Application/ApplicationContainer.tsx
Comment (nmadruga) [reply src/components/Application/ApplicationContainer.tsx:19]: NP. Putting this back.

File: src/components/Review/ReviewContainer.tsx
Comment (nmadruga) [reply src/components/Review/ReviewContainer.tsx:15]: Yep, will add that too. Thanks

File: semantic/src/site/elements/label.overrides
Comment (nmadruga) [reply semantic/src/site/elements/label.overrides:33]: Cool, didn't know about this one. Changing this now

Comment (nmadruga) [reply semantic/src/site/elements/label.overrides]: Done: 985669f

File: semantic/src/site/elements/label.overrides
Comment (nmadruga) [reply semantic/src/site/elements/label.overrides:33]: Sure. Good idea. I'll move to the `site.overrides` file

File: semantic/src/site/elements/icon.overrides
Comment (nmadruga) [reply semantic/src/site/elements/icon.overrides]: After talking in person. Done: 43e1624

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Cool, looking good. Just a few finicky things:

- I think it's quite important to get the "white" rows to have the same height as the "grey" ones, as per the design. You'll have to play around with the row margins etc that you set in grid.overrides, so it looks like this (the exact heights are not so important, just that they match each other):

- Less important, but should be an easy fix: see if you can get the Segment border colour and border radius to match the design. I *think* it should be safe to do it globally, in which case you just need to add the following variable overrides to Segment.less:
   - `@borderRadius: 8px;`
   - `@border: 1px solid #777777;`  (not 100% sure of colour here)
  You might want to do a quick check of the other pages though to make sure those overrides don't mess something else up.

**Comment by nmadruga** [comment]
> Cool, looking good. Just a few finicky things:
> 
> * I think it's quite important to get the "white" rows to have the same height as the "grey" ones, as per the design. You'll have to play around with the row margins etc that you set in grid.overrides, so it looks like this (the exact heights are not so important, just that they match each other):
>   
> * Less important, but should be an easy fix: see if you can get the Segment border colour and border radius to match the design. I _think_ it should be safe to do it globally, in which case you just need to add the following variable overrides to Segment.less:
>   
>   * `@borderRadius: 8px;`
>   * `@border: 1px solid #777777;`  (not 100% sure of colour here)
>     You might want to do a quick check of the other pages though to make sure those overrides don't mess something else up.

Thanks for this addition, I'm running out on time today. 

I have merged #600 into this now. I think this can be merged, then I create a tiny other one to make this lasts changes requested tomorrow. Sounds alright?

**Comment by nmadruga** [comment]
> * `@borderRadius: 8px;`
> * `@border: 1px solid #777777;`  (not 100% sure of colour here)
>   You might want to do a quick check of the other pages though to make sure those overrides don't mess something else up.

Done! Seems safe didn't find other places that seemed to be used it.

**Comment by CarlosNZ** [comment]
> I have merged #600 into this now. I think this can be merged, then I create a tiny other one to make this lasts changes requested tomorrow. Sounds alright?

Sure, can you either make a new issue to address this, or add it to #662 (mopping up)

**Comment by nmadruga** [comment]
> * I think it's quite important to get the "white" rows to have the same height as the "grey" ones, as per the design. You'll have to play around with the row margins etc that you set in grid.overrides, so it looks like this (the exact heights are not so important, just that they match each other)

I'm working on it - almost good : )

**Review by CarlosNZ — APPROVED** [review]
Maybe hold off merging until you approve #600?

**Comment by nmadruga** [comment]
OK - Is [PR#600](#647) ready for re-review?

**Comment by CarlosNZ** [comment]
> OK - Is [PR#600](#647) ready for re-review?

Yup

**Comment by nmadruga** [comment]
Nice! : ) I have finish my changes here. Will have another look on your PR now

**Pull request merged**
